### PR TITLE
Convert channelz registry to use Map<> instead of InlinedVector<>.

### PR DESCRIPTION
**Pull request by markdroth**
This eliminates the need for compaction.

### CODE REVIEW

File: src/core/lib/channel/channelz_registry.cc [hunk 1]
```diff
@@ -51,83 +52,38 @@ ChannelzRegistry* ChannelzRegistry::Default() {
...
-      return int(first_non_null);
-    }
-    if (uuid < target_uuid) {
-      left = first_non_null + 1;
-    } else {
-      right = true_middle - 1;
-    }
-  }
-  return direct_hit_needed ? -1 : left;
+Map<intptr_t, BaseNode*>::iterator ChannelzRegistry::FindUuidOrNext(
```
Comment (veblush) [inline src/core/lib/channel/channelz_registry.cc]: Should this end with Locked?

File: src/core/lib/channel/channelz_registry.cc [hunk 2]
```diff
@@ -51,83 +52,38 @@ ChannelzRegistry* ChannelzRegistry::Default() {
...
+    intptr_t uuid) {
+  return std::find_if(
```
Comment (veblush) [inline src/core/lib/channel/channelz_registry.cc]: Would map.lower_bound be a better choice?

File: src/core/lib/channel/channelz_registry.cc (hunk 1 above)
Comment (markdroth) [reply src/core/lib/channel/channelz_registry.cc]: Removed this method, as per the comment below.

File: src/core/lib/channel/channelz_registry.cc (hunk 2 above)
Comment (markdroth) [reply src/core/lib/channel/channelz_registry.cc]: Good idea.  I've added a `lower_bound()` method to our map implementation and changed this code to use it.

**Comment by markdroth** [comment]
@mhaidrygoog, please review the changes to the map code.  Thanks!

**Comment by markdroth** [comment]
Known issues: #17007 #19290

**Pull request merged**